insights/parsers/nftables.py

```diff
@@ -0,0 +1,166 @@
+                                    "expr": [
+                                        {"vmap": {"key": {"payload":
+                                        {"protocol": "ip", "field": "saddr"}},
+                                        "data": "@example_map"}}]
+                                },
+                                {
+                                    "family": "ip", "table": "table1", "chain": "chain1",
+                                    "expr": [
+                                        {"match": {"op": "==",
+                                        "left": {"payload": {"protocol": "tcp", "field": "dport"}},
+                                        "right": 22}}]
+                                }
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+
+    Attributes:
+        main_data (dict): A dict contains the address_family/table/chain/rule data.
+
+    Raises:
+        SkipComponent: No available nftables.
+
+    Examples:
+        >>> type(nft_obj)
+        <class 'insights.parsers.nftables.NftListRuleSet'>
+        >>> [str(key) for key in nft_obj.main_data.keys()]  # to be compatible with python2.6/2.7, transform unicode to string
+        ['ip']
+        >>> [str(key) for key in nft_obj.main_data['ip'].keys()] # to be compatible with python2.6/2.7, transform unicode to string
+        ['table1']
+        >>> 'chain1' in nft_obj.chains('ip', 'table1')
+        True
+        >>> len(nft_obj.rules('ip', 'table1', 'chain1'))
+        2
+
+    """
+
+    def parse_content(self, content):
+        """
+        There are a lot of other types in nftables besides "chain/rule", like "map", "set", "quota".
+        But they are not extracted now since there are no rules dependent on them. If needed,
+        we can extract them later.
+        """
+        super(NftListRuleSet, self).parse_content(content)
+        self.main_data = defaultdict(dict)
+        for item in self.data['nftables']:
+            for key, value in item.items():
+                address_name = value.get('family')
+                table_name = value.get('table')
+                if key == 'table':
+                    table_name = value['name']
+                    self.main_data[address_name][table_name] = defaultdict(dict)
+                elif key == 'chain':
+                    chain_name = value['name']
+                    self.main_data[address_name][table_name]['chains'][chain_name] = {}
+                    self.main_data[address_name][table_name]['chains'][chain_name]['attrs'] = value
+                    self.main_data[address_name][table_name]['chains'][chain_name]['rules'] = []
+                elif key == 'rule':
+                    chain_name = value['chain']
+                    self.main_data[address_name][table_name]['chains'][chain_name]['rules'].append(value)
+        self.main_data = dict(self.main_data)
```

Comment:
@huali027 For the `dict` converting,  with this top level converting, the inner level dict `self.main_data[address_name][table_name]` is still `defaultdict` type, it also requires a direct `dict` converting.

Reply:
@JoySnow Please help double confirm.